scripts/training/bulletproof_training_cell.py

```
@@ -363,10 +425,11 @@ def forward(self, input_ids, attention_mask):
 # Step 9: Save results
 results = {
     'best_f1': best_f1,
-    'num_labels': num_labels,
+    'num_labels': TOTAL_LABELS,
     'target_achieved': best_f1 >= 0.7,
     'go_samples': len(go_texts),
-    'journal_samples': len(journal_texts)
+    'journal_samples': len(journal_texts),
+    'emotion_mapping': emotion_mapping
```

Ruff (0.12.2)

Undefined name `emotion_mapping`

(F821)